src/apps/DesktopAppPage.vala

```diff
@@ -400,5 +563,470 @@ namespace Ilia {
+            actions_view.row_activated.connect(on_action_activated);
+            
+            // Add direct key press handler to the actions view
+            actions_view.key_press_event.connect((key) => {
+                if (key.keyval == KEY_CODE_ENTER) {
+                    stdout.printf("Direct Enter key press in actions_view\n");
+                    launch_selected_action();
+                    return true;
+                }
+                return false;
+            });
+            
+            var scrolled = new Gtk.ScrolledWindow(null, null);
+            scrolled.add(actions_view);
+            scrolled.set_size_request(200, 250);
+            actions_popover.add(scrolled);
+            // Always reset state and focus when popover is closed
+            actions_popover.closed.connect(() => {
+                actions_popover_visible = false;
+                item_view.grab_focus();
+                // Destroy popover to avoid stale state
+                if (actions_popover != null) {
+                    actions_popover.destroy();
+                    actions_popover = null;
+                    actions_view = null;
+                    actions_model = null;
+                }
+            });
+        }
+        
+        // Launch the selected action
+        private void launch_selected_action() {
+            Gtk.TreeSelection selection = actions_view.get_selection();
+            Gtk.TreeModel model;
+            Gtk.TreeIter iter;
+            
+            if (!selection.get_selected(out model, out iter)) {
+                hide_app_actions();
+                return;
+            }
+            
+            // Get action ID
+            if (current_app_info == null) {
+                hide_app_actions();
+                return;
+            }
+            
+            // Use the stored app info instead of trying to get it from the selection
+            DesktopAppInfo app_info = current_app_info;
+            
+            string action_id;
+            actions_model.@get(iter, ACTION_VIEW_COLUMN_ID, out action_id);
+            
+            // Handle separator
+            if (action_id == "generic-separator") {
+                return;
+            }
+            
+            // Handle generic actions
+            if (action_id.has_prefix("generic-")) {
+                string? desktop_file_path = app_info.get_filename();
+                string wm_cmd = get_wm_cli(this.session_controller.get_wm_name());
+                
+                switch (action_id) {
+                    case "generic-favorite":
+                        // Toggle favorite status
+                        toggle_favorite(app_info);
+                        hide_app_actions();
+                        break;
+                
+                    case "generic-terminal":
+                        try {
+                            // Get the exec line from desktop file
+                            var key_file = new KeyFile();
```

Comment:
Move the logic for this into a seperate function for better readability. Also make sure you use `systemd-run` to launch the terminal instance.